src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -106,6 +108,112 @@ public class RuntimeConfigLoaderJsonDeserializerTests
             }
         }
 
+        /// <summary>
+        /// Tests the parsing of the runtime configuration with environment variables based on replaceEnvVar parameter.
+        /// Verifies the below five different cases:
+        /// 1. datasource options has a boolean value,
+        /// 2. datasource options has a string value,
+        /// 3. datasource options is empty,
+        /// 4. datasource options is null,
+        /// 5. datasource options is not given.
+        /// </summary>
+        /// <param name="replaceEnvVar">A boolean indicating whether to replace environment variables in the configuration.</param>
+        [DataTestMethod]
+        [DataRow(false, "mssql", DisplayName = "Do not replace environment variables containing boolean values.")]
+        [DataRow(true, "mssql", DisplayName = "Replace environment variables containing boolean values.")]
+        [DataRow(false, "cosmosdb_nosql", DisplayName = "Do not replace environment variables containing string values.")]
+        [DataRow(true, "cosmosdb_nosql", DisplayName = "Replace environment variables containing string values.")]
+        [DataRow(false, "mysql", DisplayName = "Do not replace environment variables when datasource option is empty.")]
+        [DataRow(true, "mysql", DisplayName = "Replace environment variables when datasource option is empty.")]
+        [DataRow(false, "postgresql", DisplayName = "Do not replace environment variables when datasource option is null.")]
+        [DataRow(true, "postgresql", DisplayName = "Replace environment variables when datasource option is null.")]
+        [DataRow(false, "dwsql", DisplayName = "Do not replace environment variables when datasource option is not given.")]
+        [DataRow(true, "dwsql", DisplayName = "Replace environment variables when datasource option is not given.")]
+        public void TestConfigParsingWithEnvVarReplacement(bool replaceEnvVar, string databaseType)
+        {
+            // Act
+            SetEnvironmentVariablesFromDictionary(_environmentFileContentDict);
+            string configWithEnvVar = _configWithVariableDataSource.Replace("{0}", GetDataSourceConfigForGivenDatabase(databaseType));
+
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(
```

Comment:
nit: have single operation performing the `TryParseConfig()` and save the result.

Then you can have a different line perform the assert in the `// Assert` section.